### ISSUE TITLE
Plotting with projection=wcs broken with matplotlib 3.0.0

### ISSUE THREAD

**Issue by ivvv** [astropy/astropy#7862]
I am on `astropy 3.0.4`, `python 3.7` and `matplotlib 3.0.0`.

```
import matplotlib.pylab as plt
from astropy.io import fits
from astropy.wcs import WCS

hdu = fits.open(any_fits_image_with_wcs)
#
image = hdu['image'].data
fig = plt.figure()
wcs = WCS(hdu['image'].header)
ax = fig.add_subplot(111,projection=wcs)
ax.imshow(image.data,cmap=plt.cm.gray,origin='lower',interpolation='nearest')
```
The error is:

```
TypeError: get_tightbbox() takes 2 positional arguments but 3 were given
```
The same code works nicely in `matplotlib 2.2.2`.

I am not sure if this is the same issue as reported in #7839, but there it only mentions 

> the get_tightbbox issue that is causing the docs build to fail

nothing on plotting with projection=wcs.

**Comment by astrofrog** [comment]
This should be fixed in master - we should probably get a 3.0.5 release out shortly.

**Comment by astrofrog** [comment]
I'll close this since it's fixed in master, and will be released soon.
